src/odemis/acq/stream/_helper.py

```diff
@@ -721,7 +723,45 @@ def __init__(self, name, detector, dataflow, emitter, **kwargs):
 
         # TODO: grating/cw as VAs (from the spectrometer)
 
-    # onActive: same as the standard LiveStream (ie, acquire from the dataflow)
+        if self.light:
+            # Current channel index to be used for channel's power update
+            self._channel_idx = 0
+
+            pwr_range = tuple(r[self._channel_idx] for r in self.light.power.range)
+            # keep the current power setting for the VA
+            self.power = model.FloatContinuous(self.light.power.value[self._channel_idx], range=pwr_range,
+                                               unit=self.light.power.unit)
+            self.power.subscribe(self._onPower)
+
+    def _onPower(self, value):
+        """
+        Update the light power with the current channel value if the stream is active/playing.
+        :param value (float): current channel value
+        """
+        if self.is_active.value:
+            pwr = list(self.light.power.range[0])
+            pwr[self._channel_idx] = value
+            self.light.power.value = pwr
+
+    def _onActive(self, active):
+        if active:
+            if self.power:
```

Comment:
Still doesn't work ;-)
I'd suggest you test on a system without a "light" component to be 100% sure nothing is broken there.

Reply:
Yes you are right, I somehow assumed self.power was None.